pilot/pkg/bootstrap/server_test.go

```diff
@@ -603,20 +605,13 @@ func TestWatchDNSCertForK8sCA(t *testing.T) {
 		},
 	}
 
-	csr := &cert.CertificateSigningRequest{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "abc.xyz",
-		},
-		Status: cert.CertificateSigningRequestStatus{
-			Certificate: testcerts.ServerCert,
-		},
-	}
 	s := &Server{
 		server:                  server.New(),
 		istiodCertBundleWatcher: keycertbundle.NewWatcher(),
-		kubeClient:              kube.NewFakeClient(csr),
+		kubeClient:              kube.NewFakeClient(),
```

Comment:
why did we need to change this? The new fake logic only triggers on calls to Create/Update/Etc, not objects we pass in here?

Reply:
The 'csr' variable was being used to spoof a kubernetes response to a CSR, by putting it at index 0 on calls to list().  This means that, in test, the list() call expects to get the same CSR it has just approved, but it incidentally gets this signed CSR instead.  

Now that we have a controller to spoof signing CSRs, we no longer need to rely on the order of output in list, and we no longer need a second CSR.